src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -502,6 +505,87 @@ public OrganisationEntityResponseV2 retrieveOrganisationForV2Api(
         return new OrganisationEntityResponseV2(organisation, true, isPendingPbaRequired, false, true);
     }
 
+    @Override
+    public MultipleOrganisationsResponse retrieveOrganisationsByProfileIds(List<String> organisationProfileIds,
+                                                                           UUID searchAfter) {
+        List<String> orgTypes = convertProfileIdsToOrgTypes(organisationProfileIds);
+
+        boolean orgIdFilterProvided = !orgTypes.isEmpty();
+        boolean searchAfterProvided = searchAfter != null;
+
+        List<Organisation> organisations = new ArrayList<>();
+
+        if (orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeInAndIdGreaterThan(orgTypes, searchAfter);
+        }
+
+        if (orgIdFilterProvided && !searchAfterProvided) {
+            organisations = organisationRepository.findByOrgTypeIn(orgTypes);
+        }
+
+        if (!orgIdFilterProvided && searchAfterProvided) {
+            organisations = organisationRepository.findByIdGreaterThan(searchAfter);
+        }
+
+        if (!orgIdFilterProvided && !searchAfterProvided) {
```

Comment:
Should I make the validator fail if no filter or searchAfter is provided?
Or limit

Reply:
Or create a new select query to fetch the required columns
`"organisationIdentifier": "1234",
      "status": "ACTIVE",
      "lastUpdated": "2023-11-20T15:51:33.046Z",
      "organisationProfileIds": ["SOLICITOR_PROFILE"]`